arq/jobs.py

```diff
@@ -86,7 +86,7 @@ async def result(self, timeout: Optional[float] = None, *, pole_delay: float = 0
                 result = info.result
                 if info.success:
                     return result
-                elif isinstance(result, Exception):
+                elif isinstance(result, Exception) or isinstance(result, asyncio.CancelledError):
```

Comment:
Not a maintainer, but wouldn't: 
```suggestion
                elif isinstance(result, (Exception, asyncio.CancelledError)):
```
work too ?